### ISSUE TITLE
#addToClassPath: always returns true

### ISSUE THREAD

**Issue by marschall** [hpi-swa/graalsqueak#77]
`#addToClassPath:` always returns true even if the JAR file does not exist. I tested with 1.0.0-rc2 on macOs.

Use the following reproducer:

```
Java addToClassPath: 'a.jar'.
```

**Comment by fniephaus** [comment]
Here's the implementation:
https://github.com/hpi-swa/graalsqueak/blob/9ce025f5f891a6369881551897e39a604cd6e617/src/de.hpi.swa.graal.squeak/src/de/hpi/swa/graal/squeak/nodes/plugins/PolyglotPlugin.java#L1166-L1173

[This is the Javadoc](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/TruffleLanguage.Env.html#addToHostClassPath-com.oracle.truffle.api.TruffleFile-). We could add a check in the primitive, but it's probably better to do this on the Smalltalk level as the `PolyglotPlugin` is supposed to expose the API in an unmodified way. Maybe the primitive should even return the receiver, to ensure users don't rely on the return value.

**Comment by marschall** [comment]
Maybe we could check whether 

```
method.image.env.getTruffleFile(path)
```

there is however still a chance for a race condition

**Comment by fniephaus** [comment]
Of course, a race condition would still be possible. To be fair, you can misconfigure your class path and add non-existing file without the VM complaining, right? So maybe you should rather do a check after calling `addToClassPath:` to ensure your class is there?

**Comment by marschall** [comment]
> To be fair, you can misconfigure your class path and add non-existing file without the VM complaining, right?

On the command line? Yes you can. Though to be pedantic you can't add the the class path, only to a custom class loader, which is also what the Javadoc seems to imply.

> So maybe you should rather do a check after calling addToClassPath: to ensure your class is there?

Maybe, but the class could not be loaded for a different reason, like the superclass could not be loaded. Or you're in a generic part and don't know which class is supposed to be present.

However then the question remains what is the return value of true supposed to signify.

**Comment by fniephaus** [comment]
> Maybe, but the class could not be loaded for a different reason, like the superclass could not be loaded. Or you're in a generic part and don't know which class is supposed to be present.

That also means that a file exist check may not be sufficient (e.g. in case dependencies are missing).

>However then the question remains what is the return value of true supposed to signify.

I'm going to let the primitive return `self` for now. At the moment, the user must ensure it does what the user wants. If you have a better idea, we should discuss it with the GraalVM people. Maybe on Slack or in an issue on https://github.com/oracle/graal/issues.
